### PR TITLE
MMGProcess: Consistent remeshing for Lagrangian meshes

### PR DESCRIPTION
**Pull request by loumalouomega**
The reference mesh is used as base to remesh

### CODE REVIEW

**Comment by loumalouomega** [comment]
@RiccardoRossi PING

File: applications/MeshingApplication/custom_processes/mmg_process.cpp [hunk 1]
```diff
@@ -266,25 +266,51 @@ void MmgProcess<TDim>::InitializeMeshData()
...
+            auto it_node = nodes_array.begin() + i;
+            
+            SetNodes(it_node->X0(), it_node->Y0(), it_node->Z0(), nodes_colors[it_node->Id()], i + 1);
+            
+            bool blocked = false;
+            if (it_node->IsDefined(BLOCKED) == true)
+            {
+                blocked = it_node->Is(BLOCKED);
+            }
+            if (TDim == 3 && blocked == true)
```
Comment (RiccardoRossi) [inline applications/MeshingApplication/custom_processes/mmg_process.cpp:283]: the if TDim == 3 would be best out of the loop no?

File: applications/MeshingApplication/custom_processes/mmg_process.cpp [hunk 2]
```diff
@@ -732,6 +758,21 @@ void MmgProcess<TDim>::ExecuteRemeshing()
     /* After that we reorder nodes, conditions and elements: */
     ReorderAllIds();
     
+    /* Unmoving the original mesh to be able to interpolate */
+    if (mFramework == Lagrangian)
```
Comment (RiccardoRossi) [inline applications/MeshingApplication/custom_processes/mmg_process.cpp:762]: i think that the Enum name is misleading. If it is UpdatedLagrangian you may well interpolate on the latest configuration. If you are Total lagrangian than you may need to do what you are doing here

File: applications/MeshingApplication/custom_processes/nodal_values_interpolation_process.h [hunk 3]
```diff
@@ -138,30 +138,31 @@ class NodalValuesInterpolationProcess
...
         // We create the locator
-        BinBasedFastPointLocator<TDim> PointLocator = BinBasedFastPointLocator<TDim>(mrOriginMainModelPart);
-        PointLocator.UpdateSearchDatabase();
+        BinBasedFastPointLocator<TDim> point_locator = BinBasedFastPointLocator<TDim>(mrOriginMainModelPart);
+        point_locator.UpdateSearchDatabase();
         
         // Iterate in the nodes
-        NodesArrayType& pNode = mrDestinationMainModelPart.Nodes();
-        auto numNodes = pNode.end() - pNode.begin();
+        NodesArrayType& p_node = mrDestinationMainModelPart.Nodes();
```
Comment (RiccardoRossi) [inline applications/MeshingApplication/custom_processes/nodal_values_interpolation_process.h]: pNode is a wrong name for this. It is not a pointer, but rather an array. Call it NodeContainer or something of the sort.

also  i wouldn't use auto in numNodes (it is a simple type), and i would call the function size()

     int numNodes = rDestinationMainModelPart.Nodes().size();

**Review by RiccardoRossi — APPROVED** [review]
minor comments. i am not blocking for this, however please fix those

File: applications/MeshingApplication/custom_processes/mmg_process.cpp (hunk 2 above)
Comment (loumalouomega) [reply applications/MeshingApplication/custom_processes/mmg_process.cpp:762]: I am using the original configuration for search the nodes, the interpolated values are the ones from the last configuration

File: applications/MeshingApplication/custom_processes/mmg_process.cpp (hunk 1 above)
Comment (loumalouomega) [reply applications/MeshingApplication/custom_processes/mmg_process.cpp:283]: Yes, but i will duplicate code :(

File: applications/MeshingApplication/custom_processes/nodal_values_interpolation_process.h (hunk 3 above)
Comment (loumalouomega) [reply applications/MeshingApplication/custom_processes/nodal_values_interpolation_process.h]: Sorry, my fault

**Pull request merged**